lib/bambora/client.rb

```diff
@@ -19,20 +19,30 @@ def initialize(options = {})
       yield(self) if block_given?
     end
 
-    def_delegators :connection, :request
-
     protected
 
-    def connection
-      @connection ||= Excon.new(base_url, headers: headers)
+    def get(path:, params:, headers:)
+      connection.get(path, params, headers)
     end
 
-    def headers
-      Bambora::Headers.build(passcode: passcode, sub_merchant_id: sub_merchant_id)
+    def post(path:, body:, headers:)
+      connection.post(path, body.to_json.to_s, headers)
+    end
+
+    def delete(path:, headers:)
+      connection.delete(path) do |req|
+        req.headers = headers
```

Comment:
Was this done using a block so that you didn't have to unnecessarily set the `body` parameter in the call?

Reply:
Yeah, faraday uses positional arguments, so this would have been `connection.delete(path, {}, headers)`